tdrs-backend/scripts/db_backup.py

```diff
@@ -0,0 +1,228 @@
+    os.environ["AWS_ACCESS_KEY_ID"] = S3_ACCESS_KEY_ID
+    os.environ["AWS_SECRET_ACCESS_KEY"] = S3_SECRET_ACCESS_KEY
+
+    # Set Database connection info
+    AWS_RDS_SERVICE_JSON = json.loads(OS_ENV['VCAP_SERVICES'])['aws-rds'][0]['credentials']
+    DATABASE_PORT = AWS_RDS_SERVICE_JSON['port']
+    DATABASE_PASSWORD = AWS_RDS_SERVICE_JSON['password']
+    DATABASE_DB_NAME = AWS_RDS_SERVICE_JSON['db_name']
+    DATABASE_HOST = AWS_RDS_SERVICE_JSON['host']
+    DATABASE_USERNAME = AWS_RDS_SERVICE_JSON['username']
+
+    # write .pgpass
+    with open('/home/vcap/.pgpass', 'w') as f:
+        f.write(DATABASE_HOST+":"+DATABASE_PORT+":"+DATABASE_DB_NAME+":"+DATABASE_USERNAME+":"+DATABASE_PASSWORD)
+    os.environ['PGPASSFILE'] = '/home/vcap/.pgpass'
+
+except Exception as e:
+    print(e)
+    sys.exit(1)
+
+
+def backup_database(file_name,
+                    postgres_client,
+                    database_uri):
+    """Back up postgres database into file."""
+    """
+    :param file_name: back up file name
+    :param postgres_client: directory address for postgres application
+    :param database_uri: postgres URI
+    pg_dump -F c --no-acl --no-owner -f backup.pg postgresql://${USERNAME}:${PASSWORD}@${HOST}:${PORT}/${NAME}
+    """
+    try:
+        os.system(postgres_client + "pg_dump -Fc --no-acl -f " + file_name + " -d " + database_uri)
+        return True
+    except Exception as e:
+        logging.log(e)
+        return False
+
+
+def restore_database(file_name, postgres_client, database_uri=DATABASE_URI):
+    """Restore the database from filename."""
+    """
+    :param file_name: database backup filename
+    :param postgres_client: directory address for postgres application
+    """
+    try:
+        os.system(postgres_client + "createdb " + "-U " + DATABASE_USERNAME + " -h " + DATABASE_HOST +
+                  " -T template0 " + DATABASE_DB_NAME)
+    except Exception as e:
+        logging.log(e)
+    try:
+        os.system(postgres_client + "pg_restore --clean --no-owner --no-privileges --no-acl --create"
+                                    " -d " + database_uri + " " + file_name)
+        return True
+    except Exception as e:
+        logging.log(e)
+        return False
+
+
+def upload_file(file_name, bucket, object_name=None, region='us-gov-west-1'):
+    """Upload a file to an S3 bucket."""
+    """
+    :param file_name: file name being uploaded to s3 bucket
+    :param bucket: bucket name
+    :param object_name: S3 object name. If not specified then file_name is used
+    :param region: s3 AWS region to be used. defaults to government west
+    :return: True is file is uploaded, False if not successful
+    """
+    if object_name is None:
+        object_name = os.path.basename(file_name)
+    # upload the file
+    s3_client = boto3.client('s3', region_name=region)
+    try:
+        s3_client.upload_file(file_name, bucket, object_name)
```

Comment:
After this is uploaded, can we delete the file from the filesystem to not leave remnants around? Thanks :)